app/src/main/java/fr/free/nrw/commons/nearby/fragments/NearbyParentFragment.java

```diff
@@ -340,32 +345,43 @@ private void addCheckBoxCallback() {
 
     private void performMapReadyActions() {
         if (((MainActivity)getActivity()).activeFragment == ActiveFragment.NEARBY && isMapBoxReady) {
-            checkPermissionsAndPerformAction(() -> {
-                lastKnownLocation = locationManager.getLastLocation();
-                fr.free.nrw.commons.location.LatLng target=lastFocusLocation;
-                if(null==lastFocusLocation){
-                    target=lastKnownLocation;
-                }
-                if (lastKnownLocation != null) {
-                    final CameraPosition position = new CameraPosition.Builder()
-                            .target(LocationUtils.commonsLatLngToMapBoxLatLng(target)) // Sets the new camera position
-                            .zoom(ZOOM_LEVEL) // Same zoom level
-                            .build();
-                    mapBox.moveCamera(CameraUpdateFactory.newCameraPosition(position));
-                }
-                else if(locationManager.isGPSProviderEnabled()||locationManager.isNetworkProviderEnabled()){
-                    locationManager.requestLocationUpdatesFromProvider(LocationManager.NETWORK_PROVIDER);
-                    locationManager.requestLocationUpdatesFromProvider(LocationManager.GPS_PROVIDER);
-                    setProgressBarVisibility(true);
-                }
-                else {
-                    Toast.makeText(getContext(), getString(R.string.nearby_location_not_available), Toast.LENGTH_LONG).show();
-                }
-                presenter.onMapReady();
-                registerUnregisterLocationListener(false);
+            if(!applicationKvStore.getBoolean("NotDisplayLocationPermissionAlertBox", false) ||
```

Comment:
Would you mind calling it `DoNotAskForLocationPermission`? If that sounds OK to you.

Reply:
Sure, Changed!